pkg/controller/machineset/controller.go

```diff
@@ -139,3 +150,39 @@ func (c *MachineSetControllerImpl) getMachines(machineSet *v1alpha1.MachineSet)
 	}
 	return filteredMachines, err
 }
+
+func shouldAdopt(machineSet *v1alpha1.MachineSet, machine *v1alpha1.Machine) bool {
+	// Do nothing if the machine is being deleted.
+	if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+		glog.V(2).Infof("Skipping machine (%v), as it is being deleted.", machine.Name)
+		return false
+	}
+
+	// Machine owned by another controller.
+	if metav1.GetControllerOf(machine) != nil && !metav1.IsControlledBy(machine, machineSet) {
+		glog.V(2).Infof("Skipping machine (%v), as it is owned by someone else.", machine.Name)
```

Comment:
(nit)
Should be warning or error level as this is a problem situation that requires fixing.

Reply:
fixed.